// Copyright (c) 2025 by OpenTier GmbH
// SPDX-FileCopyrightText: 2025 OpenTier GmbH
// SPDX-License-Identifier: LGPL-3.0-or-later
//
// This file is part of OpenTier.
//
// This is free software: you can redistribute it and/or modify
// it under the terms of the GNU Lesser General Public License as
// published by the Free Software Foundation; either version 3 of the
// License, or (at your option) any later version.
//
// This is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Lesser General Public License for more details.
//
// You should have received a copy of the GNU Lesser General Public
// License along with this file.  If not, see <https://www.gnu.org/licenses/>.

syntax = "proto3";

package vehicle_commands;

// Enum to represent general on/off states
enum CommandState {
    OFF = 0;
    ON = 1;
}

enum LockState {
    UNLOCK = 0;
    LOCK = 1;
}

// Enum to represent the target of the command (e.g., lights, horn, etc.)
enum CommandTarget {
    LIGHTS = 0;
    HORN = 1;
    ENGINE = 2;
}

// Lock/Unlock command message
message LockUnlockCommand {
    LockState state = 2;
}

// General command for turning things on/off based on a target
message GeneralStateCommand {
    CommandState state = 2;
    CommandTarget target = 3;
}
